node/actors/consensus/src/leader/state_machine.rs

```diff
@@ -59,18 +59,30 @@ impl StateMachine {
         input: validator::Signed<validator::ConsensusMsg>,
     ) {
         let now = ctx.now();
-        let (label, result) = match &input.msg {
-            validator::ConsensusMsg::ReplicaPrepare(_) => (
-                metrics::ConsensusMsgLabel::ReplicaPrepare,
-                self.process_replica_prepare(ctx, consensus, input.cast().unwrap()),
-            ),
-            validator::ConsensusMsg::ReplicaCommit(_) => (
-                metrics::ConsensusMsgLabel::ReplicaCommit,
-                self.process_replica_commit(ctx, consensus, input.cast().unwrap()),
-            ),
+        let label = match &input.msg {
+            validator::ConsensusMsg::ReplicaPrepare(_) => {
+                let res = match self.process_replica_prepare(ctx, consensus, input.cast().unwrap())
+                {
+                    Err(err) => {
+                        tracing::warn!("{err:#}");
+                        Err(())
+                    }
+                    Ok(()) => Ok(()),
+                };
```

Comment:
Nit:  This can be expressed via `map_err()`.

Reply:
Done. I did it this way because some people don't like map() with side effects.